### PR TITLE
Check if document.createTextRange exists for IE in 'select all' plugin.

### LINKED ISSUE #545: Error thrown when pressing a "select all" button in a source view
## Are you reporting a feature or a bug?

Bug

## Provide detailed reproduction steps (if any)

1. Open a sample.
2. Switch to the source view.
3. Press "Select all" button.

### Expected result

Action should select whole editor's content.

### Actual result

Exception is thrown:
`SCRIPT438: Object doesn't support property or method 'createTextRange'`
`ckeditor.js (1065,311)`

## Other details

* Browser: Edge
* OS: Windows 10
* CKEditor version: 4.7.1

### PR DESCRIPTION
**Pull request by wojtekw92**
## What is the purpose of this pull request?

Bug fix
## Does your PR contain necessary tests?

`selectall` plugin test already exists.

### This PR contains

- [X] Unit tests
- [X] Manual tests

## What changes did you make?

selectall plugin is now checking if `createTextRange` method exists in IE.

Closes #545

### CODE REVIEW

File: tests/plugins/selectall/selectall.js [hunk 1]
```diff
@@ -45,6 +45,34 @@
 				acceptableResults,
 				bender.tools.html.prepareInnerHtmlForComparison( bender.tools.selection.getWithHtml( editor ), htmlMatchingOpts )
 			);
+		},
+
+		'test selectall in source view': function() {
```
Comment (f1ames) [inline tests/plugins/selectall/selectall.js]: The test look s quite complex, I would simplify it:
- The eventsRecorder is not needed, this test should check (according to its description) if `selectAll` works in source mode, no need to check for additional events.
- Editor creation can be moved to `bender.editors = { ... }`, like:
```
    editorSource: {
			name: 'test_source_mode',
			startupData: '<p>foo</p><p>bar</p>',
			config: {
				startupMode: 'source'
			}
		}
```
- ANd then test itself could be simplified to something like:
```
	var editor = this.editors.editorSource;

	editor.execCommand( 'selectAll' );

	assert.areSame( 'source', editor.mode, 'editor.mode' );
	if ( CKEDITOR.env.ie && CKEDITOR.env.version <= 8 ) {
		assert.areSame( document.selection.createRange().text.length, 20 );
	} else {
		assert.areSame( CKEDITOR.document.getActive().$.selectionStart, 0 );
		assert.areSame( CKEDITOR.document.getActive().$.selectionEnd, 20 );
	}
```

File: tests/plugins/selectall/manual/selectall.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<h2>Classic editor</h2>
+
+<div id="editor1">
+	<p>Some text</p>
+
+	<p>Some text2</p>
+</div>
+
+<script>
+	if ( ( CKEDITOR.env.ie && CKEDITOR.env.version < 11 ) || bender.tools.env.mobile ) {
```
Comment (f1ames) [inline tests/plugins/selectall/manual/selectall.html]: As this is only Edge issue we decided that test should be ignored for other browsers.

**Comment by f1ames** [comment]
@mlewand Please check if all your review requests got solved.

**Comment by mlewand** [comment]
@f1ames I'm good with it.

**Pull request merged**